File: sdks/java/src/main/java/com/boltprediction/sdk/model/FaultPatternSchema.java
package com.boltprediction.sdk.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.util.List;
import java.util.Map;
import java.time.OffsetDateTime;

/** 故障模式特征 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class FaultPatternSchema {

    @JsonProperty("trend_slope")
    private Double trendSlope;

    @JsonProperty("volatility")
    private Double volatility;

    @JsonProperty("sudden_changes")
    private Integer suddenChanges;

    @JsonProperty("min_value")
    private Double minValue;

    @JsonProperty("max_value")
    private Double maxValue;

    @JsonProperty("mean_value")
    private Double meanValue;

    public Double getTrendSlope() {
        return trendSlope;
    }

    public void setTrendSlope(Double trendSlope) {
        this.trendSlope = trendSlope;
    }

    public Double getVolatility() {
        return volatility;
    }

    public void setVolatility(Double volatility) {
        this.volatility = volatility;
    }

    public Integer getSuddenChanges() {
        return suddenChanges;
    }

    public void setSuddenChanges(Integer suddenChanges) {
        this.suddenChanges = suddenChanges;
    }

    public Double getMinValue() {
        return minValue;
    }

    public void setMinValue(Double minValue) {
        this.minValue = minValue;
    }

    public Double getMaxValue() {
        return maxValue;
    }

    public void setMaxValue(Double maxValue) {
        this.maxValue = maxValue;
    }

    public Double getMeanValue() {
        return meanValue;
    }

    public void setMeanValue(Double meanValue) {
        this.meanValue = meanValue;
    }

}

File: sdks/python/bolt_prediction_sdk/models/fault_pattern_schema.py
"""
FaultPatternSchema 模型
"""

from typing import Optional, Any, Dict, List
from datetime import datetime
from pydantic import Field

from .base import SDKBaseModel


class FaultPatternSchema(SDKBaseModel):
    """故障模式特征"""

    trend_slope: float = Field(description="趋势斜率")
    volatility: float = Field(description="波动率")
    sudden_changes: int = Field(description="骤降/突变点数量")
    min_value: float = Field(description="最小值")
    max_value: float = Field(description="最大值")
    mean_value: float = Field(description="平均值")
